src/Data/Parity/Properties.agda

```diff
@@ -0,0 +1,415 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties about signs
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Parity.Properties where
+
+open import Algebra.Bundles
+open import Data.Empty
+open import Data.Parity.Base renaming (Parity to ℙ)
+open import Data.Product using (_,_)
+open import Data.Sign as Sign using () renaming (Sign to 𝕊; + to 0𝕊; - to 1𝕊)
+open import Function hiding (Inverse)
+open import Level using (0ℓ)
+open import Relation.Binary using (Decidable; Setoid; DecSetoid)
+open import Relation.Binary.PropositionalEquality
+open import Relation.Nullary using (yes; no)
+
+open import Algebra.Structures {A = ℙ} _≡_
+open import Algebra.Definitions {A = ℙ} _≡_
+
+open import Algebra.Consequences.Propositional using (comm+distrˡ⇒distrʳ)
+open import Algebra.Morphism
+
+module ℙto𝕊 = Algebra.Morphism.Definitions ℙ 𝕊 _≡_
+module 𝕊toℙ = Algebra.Morphism.Definitions 𝕊 ℙ _≡_
+
+------------------------------------------------------------------------
+-- Equality
+
+infix 4 _≟_
+
+_≟_ : Decidable {A = ℙ} _≡_
+1ℙ ≟ 1ℙ = yes refl
+1ℙ ≟ 0ℙ = no λ()
+0ℙ ≟ 1ℙ = no λ()
+0ℙ ≟ 0ℙ = yes refl
+
+≡-setoid : Setoid 0ℓ 0ℓ
+≡-setoid = setoid ℙ
+
+≡-decSetoid : DecSetoid 0ℓ 0ℓ
+≡-decSetoid = decSetoid _≟_
+
+------------------------------------------------------------------------
+-- opposite
+
+p≢opposite[p] : ∀ p → p ≢ opposite p
+p≢opposite[p] 1ℙ ()
+p≢opposite[p] 0ℙ ()
+
+opposite-inverts : ∀ {p q} → opposite p ≡ q → p ≡ opposite q
+opposite-inverts { 1ℙ } { 0ℙ } refl = refl
+opposite-inverts { 0ℙ } { 1ℙ } refl = refl
+
+opposite-involutive : ∀ p → opposite (opposite p) ≡ p
+opposite-involutive p = sym (opposite-inverts refl)
+
+opposite-injective : ∀ {p q} → opposite p ≡ opposite q → p ≡ q
+opposite-injective {p} {q} eq = begin
+  p ≡⟨ opposite-inverts eq ⟩
```

Comment:
Sure. I have terrible trouble with my editor ('default' `emacs`) and the library when browsing other reasoning proofs, because the alignment seems off, but only sometimes, not consistently.

Reply:
I've had alignment issues in emacs because the default font used by emacs doesn't do the monospacing properly with the blackboard bold characters (e.g. `ℙ`) so if I'm using qualified names things get weird. Using another monospace font fixes them.

Reply:
Better now?